nisystemlink/clients/spec/models/__query_specs.py

```diff
@@ -0,0 +1,145 @@
+
+    ID = "ID"
+    SPEC_ID = "SPEC_ID"
+
+
+class QuerySpecificationsRequest(JsonModel):
+    """The request to query specifications."""
+
+    product_ids: List[str]
+    """IDs of the products to query the specifications for."""
+
+    take: Optional[int]
+    """Maximum number of specifications to return in the current API response.
+
+    Uses the default if the specified value is negative. The default value is `1000` specs.
+    """
+
+    continuation_token: Optional[str]
+    """Allows users to continue the query at the next specification that matches the given criteria.
+
+    To retrieve the next batch of specifications, pass the continuation token from the previous
+    batch in the next request. The service responds with the next batch of data and provides a new
+    continuation token. To paginate results, continue sending requests with the newest continuation
+    token provided in each response.
+    """
+
+    filter: Optional[str]
+    """
+    The specification query filter in Dynamic Linq format.
+
+    Allowed properties in the filter are:
+    - `specId`: String representing the SpecID of a specification.
+    - `name`: String representing the name of a specification.
+    - `category`: String representing the category of a specification.
+    - `type`: String enumeration representing the type of the specification.
+        Possible values are : PARAMETRIC, FUNCTIONAL
+    - `block`: String representing the block of a specification.
+    - `symbol`: String representing the symbol of a specification.
+    - `unit`: String representing the unit of a specification.
+    - `workspace`: String representing the ID of the workspace the specification belongs to.
+    - `createdBy`: String representing the ID of the user who created the specification.
+    - `createdAt`: ISO-8601 formatted UTC timestamp indicating when the specification was created.
+    - `updatedBy`: String representing the ID of the user who updated the specification.
+    - `updatedAt`: ISO-8601 formatted UTC timestamp indicating when the specification was updated.
+
+    See [Dynamic Linq](https://github.com/ni/systemlink-OpenAPI-documents/wiki/Dynamic-Linq-Query-Language)
+    documentation for more details.
+    """
+
+    projection: Optional[List[ProjectionEnum]]
+    """Specifies the fields to include in the returned specifications.
+
+    Fields you do not specify are excluded. Returns all fields if no value is specified.
+    """
+
+    order_by: Optional[OrderBy]
+    """Specifies the field to use to sort specifications.
+
+    By default, specifications are sorted by `ID`.
+    """
+    order_by_descending: Optional[bool]
+    """Specifies whether to return the specifications in descending order.
+
+    By default, this value is `false` and specifications are sorted in ascending order.
+    """
+
+
+class QuerySpecificationResponseObject(SpecificationBase):
+    """A single spec that matches the query."""
+
+    id: Optional[str]
```

Comment:
I did a review of these and it looks like they are correctly nullable (Optional). I think it may be true that the swagger isn't quite correct, but I think it's ok to have them optional for now. This is another one that I think we can revisit in a follow-on PR if needed.